File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventProperty.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

/**
 * An event property have all possible types
 */
case class EventProperty(
    floatVal: scala.Option[Double],
    booleanVal: scala.Option[Boolean],
    intVal: Int,
    intVals: scala.Seq[Int],
    stringVal: String
) {

    override def toString(): String = {
        scala.Seq(
            if (floatVal.isDefined) "floatVal: " + GraphQLRequestSerializer.getEntry(floatVal.get) else "",
            if (booleanVal.isDefined) "booleanVal: " + GraphQLRequestSerializer.getEntry(booleanVal.get) else "",
            "intVal: " + GraphQLRequestSerializer.getEntry(intVal),
            if (intVals != null) "intVals: " + GraphQLRequestSerializer.getEntry(intVals.asJava) else "",
            if (stringVal != null) "stringVal: " + GraphQLRequestSerializer.getEntry(stringVal) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventProperty.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

/**
 * An event property have all possible types
 */
data class EventProperty(
    val floatVal: Double?,
    val booleanVal: Boolean?,
    val intVal: Int,
    val intVals: List<Int>?,
    val stringVal: String?
) {

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        if (floatVal != null) {
            joiner.add("floatVal: " + GraphQLRequestSerializer.getEntry(floatVal))
        }
        if (booleanVal != null) {
            joiner.add("booleanVal: " + GraphQLRequestSerializer.getEntry(booleanVal))
        }
        joiner.add("intVal: " + GraphQLRequestSerializer.getEntry(intVal))
        if (intVals != null) {
            joiner.add("intVals: " + GraphQLRequestSerializer.getEntry(intVals))
        }
        if (stringVal != null) {
            joiner.add("stringVal: " + GraphQLRequestSerializer.getEntry(stringVal))
        }
        return joiner.toString()
    }
}
